### ISSUE TITLE
Interactive element within another in Navigation Treeview Example

### ISSUE THREAD

**Issue by pbb72** [w3c/aria-practices#2577]
The Navigation Treeview Example at https://www.w3.org/WAI/ARIA/apg/example-index/treeview/treeview-navigation.html shows a treeview with expandable branches.

Each branch is an anchor (an interactive element), and the branches which contain subbranches have, inside the anchor, an icon that can be clicked to expand/collapse (ie, they are also interactive) without activating the anchor (ie, opening the corresponding page).

It is my understanding that interactive elements can not be nested inside other interactive elements. (In addition, I am surprised the expand/collapse icon is not defined as a button.)

**Comment by JAWS-test** [comment]
> It is my understanding that interactive elements can not be nested inside other interactive elements

Yes, this is correct, but the special characteristic of a treeitem of a tree is that it can have 2 separate functions: Opening and closing of subordinated treeitem and selection of the current treeitem.

I.e. it is a interactive element and not nested elements. That is why it is only a treeitem and not a link and button separately.

**Comment by JAWS-test** [comment]
> In addition, I am surprised the expand/collapse icon is not defined as a button

The icon is only a visual indicator of the status of the treeitem, which is transmitted programmatically with aria-expanded.

Analogous to the icon in the drop-down list at https://www.w3.org/WAI/ARIA/apg/example-index/combobox/combobox-select-only.html. Here, too, the icon is not transmitted separately, because it only transmits the element type, which is programmatically clear through role=combobox.

**Comment by JAWS-test** [comment]
One weakness of ARIA, however, is that the markups only help blind users. A sighted user who works with the mouse and still uses the speech output will wonder why the icon at the treeitem, which can be operated separately, has no role and label. However, this seems to me to be a problem that does not only apply to the treeitem example, but is of a general nature and should be discussed in detail.

**Comment by pbb72** [comment]
> The icon is only a visual indicator of the status of the treeitem, which is transmitted programmatically with aria-expanded.

I am not sure if we understand eachother correctly. You are correct that the icon on a drop-down list is a status indicator only and does not have separate functionality.

However, the icon on the treeview I meantioned *does* have separate functionality inside the same link, it is not only a visual indicator of the status. Clicking on the icon expands the tree, without opening the corresponding page. Clicking on the title opens the page, without expanding the treeview.

**Comment by JAWS-test** [comment]
> However, the icon on the treeview I meantioned does have separate functionality inside the same link, it is not only a visual indicator of the status. Clicking on the icon expands the tree, without opening the corresponding page. Clicking on the title opens the page, without expanding the treeview.

Yes, but this is only for mouse users and not for keyboard users. Blind users, for whom ARIA and the patterns are, are keyboard users. For keyboard users, the treeitem is one control with different functions and the icon only shows the status transmitted by aria-expanded. That's why I also meant that this is a weakness of ARIA

**Comment by pbb72** [comment]
Ah I see your point now! The interaction is different for mouse and keyboard users. And as the interaction on the icon is only added for mouse users, it does not break the rule for nested interaction. Thanks!
